### PR TITLE
Add get approvals filters

### PR DESCRIPTION
**Pull request by magiccookie**
https://github.com/edenlabllc/ehealth.api/issues/2800

### CODE REVIEW

File: lib/mithril_api/search.ex [hunk 1]
```diff
@@ -3,70 +3,114 @@ defmodule Mithril.Search do
...
-        entity
-        |> get_search_query(changes)
-        |> Repo.paginate(search_params)
-      end
+  def get_token_search_query(entity, changes), do: get_search_query(entity, changes)
 
-      def search(%Ecto.Changeset{valid?: false} = changeset, _search_params, _entity) do
-        {:error, changeset}
-      end
+  def search_apps(%Ecto.Changeset{valid?: true}, params) do
```
Comment (AlexKovalevych) [inline lib/mithril_api/search.ex]: this probably should be `Mithril.AppAPI` module

File: lib/mithril_api/app_api/app_api.ex [hunk 2]
```diff
@@ -85,14 +85,19 @@ defmodule Mithril.AppAPI do
...
     |> cast(attrs, [:user_id, :client_id, :scope])
     |> unique_constraint(:user_id, name: "apps_user_id_client_id_index")
     |> validate_required([:user_id, :client_id, :scope])
   end
 
-  defp app_changeset(%AppSearch{} = app, attrs) do
-    cast(app, attrs, [:user_id, :client_id])
+  defp app_search_changeset(%AppSearch{} = app, attrs) do
+    app
+    |> cast(attrs, AppSearch.__schema__(:fields))
```
[the remark this answers is not in the document]
Comment (magiccookie) [reply lib/mithril_api/app_api/app_api.ex]: oops.. forgot...

done.

File: lib/mithril_api/search.ex [hunk 3]
```diff
@@ -3,70 +3,114 @@ defmodule Mithril.Search do
...
 
-        App
-        |> preload(:clients)
-        |> join(:inner, [a], s in subquery(apps_subquery), s.id == a.id)
-        |> Repo.paginate(params)
-      end
+  def get_search_query(entity, _changes), do: from(e in entity)
 
-      def search_apps(%Ecto.Changeset{valid?: false} = changeset, _params), do: {:error, changeset}
+  def search_token(%Ecto.Changeset{valid?: true, changes: changes}, params) do
```
[the remark this answers is not in the document]
Comment (magiccookie) [reply lib/mithril_api/search.ex]: done

File: lib/mithril_api/search.ex (hunk 1 above)
Comment (magiccookie) [reply lib/mithril_api/search.ex]: done

**Pull request merged**